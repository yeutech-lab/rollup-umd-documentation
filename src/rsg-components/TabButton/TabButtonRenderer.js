import React from 'react';
import PropTypes from 'prop-types';
import Button from 'bootstrap-styled/lib/Button';

export function TabButtonRenderer({
  name,
  onClick,
  active, // eslint-disable-line no-unused-vars
  children,
}) {
  return (
    <Button name={name} onClick={onClick}>
      {children}
    </Button>
  );
}
/* eslint-disable react/require-default-props */
TabButtonRenderer.propTypes = {
  name: PropTypes.string,
  onClick: PropTypes.func,
  active: PropTypes.bool,
  children: PropTypes.node,
};
/* eslint-enable react/require-default-props */
export default TabButtonRenderer;
