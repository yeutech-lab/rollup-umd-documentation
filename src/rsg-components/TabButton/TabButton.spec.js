import React from 'react';
import { TabButtonRenderer, styles } from './TabButtonRenderer';

const props = {
	classes: classes(styles),
	onClick: () => {},
};

it('should render a button', () => {
	const actual = shallow(<TabButtonRenderer {...props}>pizza</TabButtonRenderer>);

	expect(actual).toMatchSnapshot();
});

it('should render active styles', () => {
	const actual = shallow(
		<TabButtonRenderer {...props} active>
			pizza
		</TabButtonRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('should pass a class name to a button', () => {
	const actual = shallow(
		<TabButtonRenderer {...props} onClick={() => {}} className="foo-class">
			pizza
		</TabButtonRenderer>
	);

	expect(actual).toMatchSnapshot();
});
