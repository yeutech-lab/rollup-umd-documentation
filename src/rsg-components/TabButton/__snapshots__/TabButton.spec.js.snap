// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a class name to a button 1`] = `
<button
  className="button foo-class"
  onClick={[Function]}
  type="button"
>
  pizza
</button>
`;

exports[`should render a button 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  pizza
</button>
`;

exports[`should render active styles 1`] = `
<button
  className="button isActive"
  onClick={[Function]}
  type="button"
>
  pizza
</button>
`;
